pkg/assembler/backends/keyvalue/certifyBad.go

```diff
@@ -266,27 +422,27 @@ func (c *demoClient) CertifyBad(ctx context.Context, filter *model.CertifyBadSpe
 				if err != nil {
 					return nil, err
 				}
-				out, err = c.addCBIfMatch(ctx, out, filter, link)
+				cb, err := c.CBIfMatch(ctx, filter, link)
 				if err != nil {
 					return nil, gqlerror.Errorf("%v :: %v", funcName, err)
 				}
+
+				out = append(out, cb)
 			}
 		}
 	}
 	return out, nil
 }
 
-func (c *demoClient) addCBIfMatch(ctx context.Context, out []*model.CertifyBad,
-	filter *model.CertifyBadSpec, link *badLink) (
-	[]*model.CertifyBad, error) {
+func (c *demoClient) CBIfMatch(ctx context.Context, filter *model.CertifyBadSpec, link *badLink) (*model.CertifyBad, error) {
 
 	if filter != nil {
 		if noMatch(filter.Justification, link.Justification) ||
 			noMatch(filter.Collector, link.Collector) ||
 			noMatch(filter.Origin, link.Origin) ||
 			noMatch(filter.DocumentRef, link.DocumentRef) ||
 			filter.KnownSince != nil && filter.KnownSince.After(link.KnownSince) {
-			return out, nil
+			return nil, nil
```

Comment:
this returns nil now, so need to be checked wherever its used.

Reply:
I am now checking it